app/src/core/kotlin/adblocker/ActiveWidget.kt

```diff
@@ -332,16 +360,23 @@ class UpdateWidgetService : Service() {
             views.setViewVisibility(R.id.widget_dns, View.GONE)
         }
 
+        if (data.vpn) {
+            views.setViewVisibility(R.id.widget_gateway, View.VISIBLE)
+        } else {
+            views.setViewVisibility(R.id.widget_gateway, View.GONE)
+        }
+
 //        views.setInt(R.id.widget_root, "setBackgroundColor", data.alpha shl 24 or 0x00262626)
 
         val intent = Intent(this, ActiveWidgetProvider::class.java)
         intent.action = AppWidgetManager.ACTION_APPWIDGET_UPDATE
         intent.putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, data.id)
         intent.putExtra("changeBlokadaState", true)
-        val pendingIntent = PendingIntent.getBroadcast(this.applicationContext,
-                0, intent, PendingIntent.FLAG_UPDATE_CURRENT)
+        var pendingIntent = PendingIntent.getBroadcast(this.applicationContext,
+                802099703, intent, PendingIntent.FLAG_UPDATE_CURRENT)
```

Comment:
Does this constant mean anything?

Reply:
Just some random 32 bit value for the Indent-ID to prevent collisions. Maybe I should add a comment. ^^